app_dart/lib/src/foundation/utils.dart

```diff
@@ -103,6 +103,39 @@ Future<List<Map<String, dynamic>>> getRepoBuilders(HttpClientProvider branchHttp
       .toList();
 }
 
+/// Returns a LUCI [builder] list that covers changed [files].
+///
+/// [builders]: enabled luci builders.
+/// [files]: changed files in corresponding PRs.
+///
+/// [builder] is with format:
+/// {
+///   "name":"yyy",
+///   "repo":"flutter",
+///   "taskName":"zzz",
+///   "enabled":true,
+///   "run_if":["a/b/", "c/d/**"]
+/// }
+///
+/// [file] is based on repo root: `a/b/c.dart`.
+Future<List<Map<String, dynamic>>> getFilteredBuilders(List<Map<String, dynamic>> builders, List<String> files) async {
```

Comment:
It would be really helpful to fix the unreadable, non-type-safe Map<String, dynamic> everywhere with a `@JsonSerializable` annotated class.

Reply:
+1 a list of map of maps is prone to errors.

Reply:
Agree. Updating to use annotated class will affect quite some files. Created TODO (https://github.com/flutter/flutter/issues/64286) to update that in a separate PR.